lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1355,6 +1344,48 @@ def compute_timestamp(resource_type, record, time)
         ts_secs = timestamp.tv_sec
         ts_nanos = timestamp.tv_nsec
       end
+      ts_secs = begin
+                  Integer ts_secs
+                rescue ArgumentError, TypeError
+                  ts_secs
+                end
+      ts_nanos = begin
+                   Integer ts_nanos
+                 rescue ArgumentError, TypeError
+                   ts_nanos
+                 end
+
+      # Adjust timestamps from the future.
+      if ts_secs.is_a?(Integer) && ts_nanos.is_a?(Integer)
+        parsed_timestamp = Time.at(ts_secs + ts_nanos / 1_000_000_000.0)
+        end_of_year = Time.utc(Time.now.year, 12, 31, 23, 59, 59.999)
```

Comment:
Careful of timezones. Is the timestamp you're comparing with local or UTC? If the former, shouldn't you be computing the local end of the year?